launch/launch/actions/execute_process.py

```diff
@@ -382,6 +383,15 @@ def _shutdown_process(self, context, *, send_sigint):
         if self.__shutdown_received:
             # Do not handle shutdown more than once.
             return None
+
+        if self.process_details is None or self._subprocess_transport is None:
+            # Skip shutting down if the process is not ready
+            context.register_event_handler(
+                OnProcessStart(
```

Comment:
This doesn't seem to be the correct approach to me.
In this case, we don't need to do anything. Only setting `self.__shutdown_received` to `True` and returning.